memgpt/schemas/block.py

```diff
@@ -0,0 +1,51 @@
+from typing import List, Optional, Union
+
+from pydantic import BaseModel, Field, validator
+
+# block of the LLM context
+
+
+class Block(BaseModel):
+    """Block of the LLM context"""
+
+    description: Optional[str] = Field(None, description="Description of the block.")
+    limit: int = Field(2000, description="Character limit of the block.")
+    value: Optional[Union[List[str], str]] = Field(None, description="Value of the block.")
+
+    def __setattr__(self, name, value):
+        """Run validation if self.value is updated"""
+        super().__setattr__(name, value)
+        if name == "value":
+            # run validation
+            self.__class__.validate(self.dict(exclude_unset=True))
+
+    @validator("value", always=True)
+    def check_value_length(cls, v, values):
+        if v is not None:
+            # Fetching the limit from the values dictionary
+            limit = values.get("limit", 2000)  # Default to 2000 if limit is not yet set
+
+            # Check if the value exceeds the limit
```

Comment:
ready for code golf? ⛳ 
you can duck type this. 

```python
def total_len(values):
    return sum([len(v) for v in values])

a = 'this is a string'
b = ['this is a string','this is also a string']
>> total_len(a)
16
>> total_len(b)
37
```
note you don't need to worry about handling types that aren't either a string or a list of strings, because you typed them in the object - that's pydantic's job, to make sure that by the time your business logic gets called the typing has been checked :)